core/workspace.js

```diff
@@ -523,8 +523,8 @@ Blockly.Workspace.prototype.fireChangeListener = function(event) {
   if (event.recordUndo) {
```

Comment:
Is it worth also making this an && this.MAX_UNDO > 0?

Reply:
Done, though it should be >= rather than > since 0 is valid.

Reply:
If it's 0 we shouldn't add anything to the stack.

Reply:
Yes, but we can't short-circuit as you suggest since we do need to empty the stack.

Reply:
Ah, good point. I'd been assuming this was only set at initialization when the undo stack would be empty anyway. In that case >= is still needed to keep from crashing on negative values.